### ISSUE TITLE
W1031: SSM parameter name does not match regex when using !Sub

### ISSUE THREAD

**Issue by mtschneiders** [aws-cloudformation/cfn-lint#4061]
### CloudFormation Lint Version

1.32.1

### What operating system are you using?

Ubuntu

### Describe the bug

Rule W1031 raises an error on a valid SSM parameter definition: W1031 {'Fn::Sub': '/${ApplicationId}/Settings/BaseUrl'} does not match '^((?![aA][wW][sS]|[sS]{2}[mM])[\\w.-]+|\\/(?![aA][wW][sS]|[sS]{2}[mM])[\\w.-]+(\\/[\\w.-]+))$' when 'Fn::Sub' is resolved

### Expected behavior

Rule W1031 does not raise an error on a valid SSM parameter definition when using !Sub.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31

Parameters:

  ##########
  # Common #
  ##########
  ApplicationId:
    Default: app
    Type: String
    
Resources:

  LegacyIntegrationSettingsBaseUrl:
    Type: AWS::SSM::Parameter
    Properties:
      Name: !Sub /${ApplicationId}/Settings/BaseUrl
      Type: String
      Value: somevalue

```
